modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4986,7 +4986,8 @@ else if (StringUtil.equals(
 			if (!entryValueString.isEmpty()) {
 				long value = GetterUtil.getLong(entryValue);
 
-				if (!StringUtil.equals(
+				if ((value == 0) &&
+					!StringUtil.equals(
```

Comment:
The same situation happens with `ObjectFieldConstants.DB_TYPE_INTEGER`. Can you apply your fix there and add a test for it, too?

Reply:
Done!  (๑ > ᴗ < ๑)